instrumentation/tomcat/tomcat-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/tomcat/common/TomcatHelper.java

```diff
@@ -32,7 +33,9 @@ public boolean shouldStart(Context parentContext, Request request) {
   }
 
   public Context start(Context parentContext, Request request) {
-    return instrumenter.start(parentContext, request);
+    Context context = instrumenter.start(parentContext, request);
+    request.setAttribute(HttpServerTracer.CONTEXT_ATTRIBUTE, context);
```

Comment:
Is this the actual fix?

Reply:
this would be nice if it leads to unifying app server behavior a bit, I created #4347 to revisit the idea